### ISSUE TITLE
Incorrect parsing of a typedef

### ISSUE THREAD

**Issue by kdschlosser** [doxygen/doxygen#9944]
**Describe the bug**
In my use case Doxygen is being used for it's XML output. I am building the documentation using Sphinx and Breathe. I had originally thought the problem was in Breathe but upon further investigation I discovered it is a problem in Doxygen.

This is the error I am getting from Sphinx/Breathe

```
If typedef-like declaration:
  Type must be either just a name or a typedef-like declaration.
  If just a name:
    Error in declarator or parameters-and-qualifiers
    Invalid C++ declaration: Expected identifier in nested name, got keyword: void [error at 4]
      void() lv_lru_free_t (void *v)
      ----^
```

This is the declaration of the typedef

```c
typedef void (lv_lru_free_t)(void * v);
```

The signatures don't match `typedef void (lv_lru_free_t)(void * v);` != `void() lv_lru_free_t (void *v)`

I dove into the xml files and located this typedef. Below is the elements that are for this typedef

``` xml
<memberdef kind="typedef" id="lv__lru_8h_1abe348580acc0f079a2014ec2d3cc2650" prot="public" static="no">
        <type>void()</type>
        <definition>typedef void() lv_lru_free_t(void *v)</definition>
        <argsstring>(void *v)</argsstring>
        <name>lv_lru_free_t</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="lvgl/src/misc/lv_lru.h" line="42" column="9" bodyfile="lvgl/src/misc/lv_lru.h" bodystart="42" bodyend="-1"/>
      </memberdef>
```

This is when I realized the problem is with Doxygen. You can clearly see the `type` and `definition` elements are incorrect.

When I change the xml to read
``` xml
<memberdef kind="typedef" id="lv__lru_8h_1abe348580acc0f079a2014ec2d3cc2650" prot="public" static="no">
        <type>void</type>
        <definition>typedef void(lv_lru_free_t)(void *v)</definition>
        <argsstring>(void *v)</argsstring>
        <name>lv_lru_free_t</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="lvgl/src/misc/lv_lru.h" line="42" column="9" bodyfile="lvgl/src/misc/lv_lru.h" bodystart="42" bodyend="-1"/>
      </memberdef>
```

the documentation builds without the error.

**Expected behavior**
To have the typedef properly parsed.

**Screenshots**
N/A

**To Reproduce**
I don't have anything at the moment that is a self contained way of reproducing the problem. In the next couple of days I will be pushing a commit to a repository to handle this issue. I will make another branch where the problem is able to be seen.

**Version**
* Doxygen: 1.9.6 
* Sphinx: 6.1.3
* Breathe: 4.35.0
* CPython: 3.8.10
* Windows 11 x64: build 22621

**Stack trace**
N/A

**Additional context**
If there is anything additional I can provide just ask and I will do my best to fulfill the request.

**Comment by doxygen** [comment]
@kdschlosser Doxygen has some issues with redundant braces as is also listed [here](https://www.doxygen.nl/manual/trouble.html)

Instead of

```c++
typedef void (lv_lru_free_t)(void * v);
```
I suggest to use the equivalent

```c++
typedef void lv_lru_free_t(void * v);
```

**Comment by Albrecht-S** [comment]
I confirm that the workaround given above in https://github.com/doxygen/doxygen/issues/9944#issuecomment-1481862499 works but it is what it is: a workaround. Doxygen should parse valid syntax correctly and output correct code in the documentation. See #10295.

**Comment by albert-github** [comment]
As far as I can tell the fix:
[issue](https://github.com/doxygen/doxygen/commit/c29b0e69b907ec1ae2bdee01498fdb46cb8a6989) https://github.com/doxygen/doxygen/issues/10295 [typedef of a function incorrectly output in HTML and PDF](https://github.com/doxygen/doxygen/commit/c29b0e69b907ec1ae2bdee01498fdb46cb8a6989) also fixes this problem.

@kdschlosser please check (when OK please don't close the issue as this will be done at the moment of an official release)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
